pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -299,16 +295,10 @@ func newTXTRecordSet(fqdn, value string, ttl int) *route53.ResourceRecordSet {
 // The given error must not be nil. This function must be called everywhere
 // we have a non-nil error coming from an aws-sdk-go func.
 func removeReqID(err error) error {
-	// NOTE(mael): I first tried to unwrap the RequestFailure to get rid of
-	// this request id. But the concrete type requestFailure is private, so
-	// I can't unwrap it. Instead, I recreate a new awserr.baseError. It's
-	// also a awserr.Error except it doesn't have the request id.
-	//
-	// Also note that we do not give the origErr to awserr.New. If we did,
-	// err.Error() would show the origErr, which we don't want since it
-	// contains a request id.
-	if e, ok := err.(awserr.RequestFailure); ok {
-		return awserr.New(e.Code(), e.Message(), nil)
+	responseError := &awshttp.ResponseError{}
+	if errors.As(err, &responseError) {
+		// remove the request id from the error message
+		responseError.RequestID = "<REDACTED>"
 	}
```

Comment:
Don't you need to return the modified error here?
Is there a unit test for this error redaction feature?

Reply:
I made the following change to the test to demonstrate what I mean:

```diff
$ git diff
diff --git a/pkg/issuer/acme/dns/route53/route53_test.go b/pkg/issuer/acme/dns/route53/route53_test.go
index fcde37cb1..8c224761e 100644
--- a/pkg/issuer/acme/dns/route53/route53_test.go
+++ b/pkg/issuer/acme/dns/route53/route53_test.go
@@ -315,8 +315,8 @@ func Test_removeReqID(t *testing.T) {
        }{
                {
                        name:    "should replace the request id with a static value to keep the message stable",
-                       err:     &awshttp.ResponseError{RequestID: "SOMEREQUESTID", ResponseError: newResponseError()},
-                       wantErr: &awshttp.ResponseError{RequestID: "<REDACTED>", ResponseError: newResponseError()},
+                       err:     fmt.Errorf("simulated error: %w", &awshttp.ResponseError{RequestID: "SOMEREQUESTID", ResponseError: newResponseError()}),
+                       wantErr: fmt.Errorf("simulated error: %w", &awshttp.ResponseError{RequestID: "<REDACTED>", ResponseError: newResponseError()}),
                },
                {
                        name:    "should do nothing if no request id is set",
```


Got the following test failure:

```
$ go test ./pkg/issuer/acme/dns/route53/...
--- FAIL: Test_removeReqID (0.00s)
    --- FAIL: Test_removeReqID/should_replace_the_request_id_with_a_static_value_to_keep_the_message_stable (0.00s)
        route53_test.go:337:
                Error Trace:    /home/richard/projects/cert-manager/cert-manager/pkg/issuer/acme/dns/route53/route53_test.go:337
                Error:          Not equal:
                                expected: "simulated error: https response error StatusCode: 0, RequestID: <REDACTED>, foo"
                                actual  : "simulated error: https response error StatusCode: 0, RequestID: SOMEREQUESTID, foo"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -simulated error: https response error StatusCode: 0, RequestID: <REDACTED>, foo
                                +simulated error: https response error StatusCode: 0, RequestID: SOMEREQUESTID, foo
                Test:           Test_removeReqID/should_replace_the_request_id_with_a_static_value_to_keep_the_message_stable
FAIL
FAIL    github.com/cert-manager/cert-manager/pkg/issuer/acme/dns/route53        0.858s
FAIL
```

Reply:
With `%w` ?

Reply:
I'll just return the new ResponseError to fix this issue.

Reply:
EDIT: I noticed I lost too much info about the error, instead I just updated the unit test to reflect a realistic error message (without error message caching).

Reply:
I updated the function comment.